packages/client/hmi-client/src/components/workflow/ops/model-config/tera-model-config-drilldown.vue

```diff
@@ -301,8 +294,56 @@ const calibratedConfigObservables = computed<Observable[]>(() =>
 	}))
 );
 
-const sanityCheckErrors = ref<string[]>([]);
-const isSaveDisabled = computed(() => knobs.value.transientModelConfig.name === '');
+// Save button is disabled if the model configuration name is empty, the values have changed, or the configuration is the same as the original
+const isSaveDisabled = computed(
+	() =>
+		knobs.value.transientModelConfig.name === '' ||
+		isEqual(originalConfig, knobs.value.transientModelConfig) ||
+		haveValuesChanged.value
+);
+
+// a property to check if the values have changed (not metadata such as name, description, sources...)
+const haveValuesChanged = computed(() => {
+	const modelConfig = cloneDeep(knobs.value.transientModelConfig);
```

Comment:
Consider
- Moving this function out into a util, so we can unit test it
- Using cache maps so the lookup is more efficient and not `O(n^2)` , cause we have stratified models
- Observables

Reply:
Good points, all of the above has been added